explainaboard/feature.py

```diff
@@ -278,350 +18,123 @@ class BucketInfo:
-            return [
-                name.strip() for name in f.read().split("\n") if name.strip()
-            ]  # Filter empty names
+    def __post_init__(self):
+        self._type: str = self.__class__.__name__
 
 
 @dataclass
-class Sequence:
-    """Construct a list of feature from a single type or a dict of types.
-    Mostly here for compatiblity with tfds.
-    """
+class Sequence(FeatureType):
+    feature: FeatureType = field(default_factory=FeatureType)
 
-    feature: Any
-    length: int = -1
-    id: Optional[str] = None
-    is_bucket: bool = False
-    require_training_set: bool = False
-    # Automatically constructed
-    dtype: ClassVar[str] = "list"
-    pa_type: ClassVar[Any] = None
-    _type: str = field(default="Sequence", init=False, repr=False)
+    def __post_init__(self):
+        super().__post_init__()
+        self.dtype = "list"
 
 
 @dataclass
-class Set:
-    feature: dict
+class Set(FeatureType):
+    feature: dict[str, FeatureType] = field(default_factory=dict)
 
-    dtype: ClassVar[str] = "dict"
-    is_bucket: bool = False
-    require_training_set: bool = False
-    is_pre_computed: bool = False
-    bucket_info: Optional[BucketInfo] = None
-    _type: str = field(default="Set", init=False, repr=False)
-    id: Optional[str] = None
-    pa_type: ClassVar[Any] = None
+    def __post_init__(self):
+        super().__post_init__()
+        self.dtype = "dict"
 
 
 @dataclass
-class Position:
+class Position(FeatureType):
     positions: Optional[list] = None
-    dtype: ClassVar[Union[str, object]] = Any  # TODO(odashi): avoid using Any here.
-    is_bucket: bool = False
-    require_training_set: bool = False
-    is_pre_computed: bool = False
-    bucket_info: Optional[BucketInfo] = None
-    _type: str = field(default="Position", init=False, repr=False)
-    id: Optional[str] = None
-    pa_type: ClassVar[Any] = None
 
-
-@dataclass
-class Span:
-    dtype: ClassVar[Union[str, object]] = Any  # TODO(odashi): avoid using Any here.
-    is_bucket: bool = False
-    require_training_set: bool = False
-    is_pre_computed: bool = False
-    bucket_info: Optional[BucketInfo] = None
-    _type: str = field(default="Span", init=False, repr=False)
-    id: Optional[str] = None
-    pa_type: ClassVar[Any] = None
+    def __post_init__(self):
+        super().__post_init__()
+        self._type: str = "Position"
 
 
 @dataclass
-class Value:
-    """
-    The Value dtypes are as follows:
-    null
-    bool
-    int8
-    int16
-    int32
-    int64
-    uint8
-    uint16
-    uint32
-    uint64
-    float16
-    float32 (alias float)
-    float64 (alias double)
-    timestamp[(s|ms|us|ns)]
-    timestamp[(s|ms|us|ns), tz=(tzstring)]
-    binary
-    large_binary
-    string
-    large_string
-    """
+class Value(FeatureType):
 
-    dtype: str  # must be initialized when created
-    description: Optional[str] = None
     max_value: Optional[float | int] = None
```

Comment:
Could you add a docstring describing these values, particularly to indicate whether `max_value` is inclusive or exclusive?